lib/application.js

```diff
@@ -0,0 +1,180 @@
+var getNextLink = require("./util").getNextLink;
+
+/**
+ * Application
+ * @constructor
+ * @param {Object} client Catapult client
+ */
+
+var Application = function (client) {
+	/**
+	 * List the user's applications
+	 * @param {Object} params Parameters for filtering applications.
+	 * @param {Number} [params.size] The maximum number of applications returned by
+	 * the query per page (Max size: 1000).
+	 * @param {Function} [callback] A callback for the list of applications.
+	 * @return {Array.<ApplicationResponse>} A promise for the list of applications, has a getNextPage
+	 * function if the number of applications returned by the query exceeds the page size.
+	 */
+	this.list = function (params, callback) {
+		var self = this;
+		return client.makeRequest({
+			path   : "applications",
+			method : "GET",
+			qs     : params
+		})
+		.then(function (response) {
+			var applicationListResponse = {
```

Comment:
I know it's not required but I'd say defaulting all the variables you plan to add to this object is a good practice, so add `hasNextPage` and `getNextPage` and what those should be defaulted to should they not exist.